File: leet_code_solutions.py/27.py

# Runtime: 28 ms, faster than 92.68% of Python3 online submissions for Remove Element.
# Memory Usage: 14.2 MB, less than 47.40% of Python3 online submissions for Remove Element.

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        
        index=0
        for i in range(len(nums)):
            if nums[i]!=val:
                nums[index]=nums[i]
                index+=1
        return index

File: leet_code_solutions.js/27.js
// solved
// Runtime: 82 ms, faster than 25.49% of JavaScript online submissions for Remove Element.
// Memory Usage: better than 92.83% of JS online submissions for Remove Element
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
  var counter = 0;
  while (counter < nums.length) {
    if (nums[counter] === val) {
      nums.splice(counter, 1);
    } else {
      counter++;
    }
  }
  return nums.length;
};
